h2o-algos/src/main/java/hex/pca/PCA.java

```diff
@@ -334,30 +327,31 @@ public void computeImpl() {
             throw new IllegalArgumentException("Found validation errors: " + validationErrors());
           }
 
-          // Compute SVD of Gram A'A/n using JAMA library
-          // Note: Singular values ordered in weakly descending order by algorithm
+          // Compute SVD of Gram A'A/n using netlib-java (MTJ) library
           _job.update(1, "Calculating SVD of Gram matrix locally");
-          Matrix gramJ = _wideDataset ? new Matrix(ogtsk._gram.getXX()) : new Matrix(gtsk._gram.getXX());
-          SingularValueDecomposition svdJ = gramJ.svd();
-          _job.update(1, "Computing stats from SVD");
-          // correct for the eigenvector by t(A)*eigenvector for wide dataset
-          if (_wideDataset) {
-            double[][] eigenVecs = transformEigenVectors(dinfo, svdJ.getV().getArray());
-            computeStatsFillModel(model, dinfo, svdJ.getSingularValues(), eigenVecs, gram, model._output._nobs);
-          } else {
-            computeStatsFillModel(model, dinfo, svdJ, gram, model._output._nobs);
+          double[][] gramMatrix;
+          gramMatrix = _wideDataset ? ogtsk._gram.getXX() : gtsk._gram.getXX();
+          PCAInterface svd = null;
+          svd = PCAImplementationFactory.createSVDImplementation(gramMatrix, _parms.getSvdImplementation());
+          assert svd != null;
+          double[][] rightEigenvectors = svd.getPrincipalComponents();
```

Comment:
renaming a variable is unnecessary change (eigenVecs renamed to rightEigenvectors).  The less change you make to the code, the faster the code review will be.

Reply:
Hi @wendycwong !
This is a completely new piece of code, since I switched from hard-coded Jama to more flexible modular way to switch between PCA implementations (e.g. Jama, netlib-java, Intel DAAL...). Hence, I find it justified to rename to a more suiting name. For better future clarity.

I believe it's reasonable to leave it to the more expressive name of `rightEigenvectors`.

Thanks for your comment!